Add insecure flag in API and CLI procedure (#2012)
theforeman/foreman-documentation: guides/common/modules/snip_host-registration-steps.adoc

----
# hammer host-registration generate-command
----
+

> If your hosts do not trust the SSL certificate of {ProjectServer}, add the `--insecure` flag to the registration command.

Reviewer: I'd also say what it does:
```suggestion
If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `--insecure` flag to the registration command.
````

Author: Agree and Addressed :+1:

> If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `--insecure` flag to the registration command.

Reviewer 2: "can be disabled" > "you can disable"

+
[options="nowrap" subs="+quotes,attributes"]
----

> # hammer host-registration generate-command

Reviewer 3: ```suggestion
# hammer host-registration generate-command \
```

> --insecure 1

Reviewer 3: should this be `--insecure true`? I guess both work as far as the API is concerned, but using a true (lol) boolean seems more correct here?

----
endif::[]
ifdef::katello,satellite,orcharhino[]
[options="nowrap" subs="+quotes,attributes"]
----
# hammer host-registration generate-command \
--activation-keys "_My_Activation_Key_"
----
+

> If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `insecure` flag to the registration command.

Author: Apology. Missed it.

+
[options="nowrap" subs="+quotes,attributes"]
----
# hammer host-registration generate-command \

> --activation-keys "_My_Activation_Key_"

Reviewer 3: ```suggestion
--activation-keys "_My_Activation_Key_" \
```

-d '{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"] }}'
----
+
[NOTE]
====

> If your hosts do not trust the SSL certificate of {ProjectServer}, add the `--insecure` flag to the registration command.

Reviewer: I think this only applied to the Hammer command. For the API request it needs to be inside the JSON data.

The whole idea is that you have:
1. Client -> Foreman: request command
2. Foreman -> Client: return command
3. Client -> Host: run command
4. Host -> Foreman: register itself

If you add `--insecure` to `curl`, that's step one, but you want it to be done in step 4, so you must request Foreman in step one to render the correct flag in step 2.

The original PR was the correct way to do that, so the Bugzilla report has the correct instructions.

Author: Means, we should go with this commit https://github.com/theforeman/foreman-documentation/commit/3a20d632ca9b1f18e44427a3958ee2bb46f20d0d ?

Reviewer: That commit unconditionally specifies the parameter. I was suggesting to:
* For the CLI procedure: by default run without `--insecure 1`, but then document that it is an option. So move the note you have here to the CLI procedure.
* For the API procedure: provide examples where `insecure` is specified.

Katello (note compared to https://github.com/theforeman/foreman-documentation/commit/3a20d632ca9b1f18e44427a3958ee2bb46f20d0d it doesn't quote `true`):
```json
{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"], "insecure": true }}
```

Non-Katello:
```json
{ "registration_command": { "insecure": true }}
```

Reviewer: I can now see that `For the API procedure: provide examples` was ambiguous. What I meant was to first document the examples as they are today and then document you can make it insecure and then provide additional examples where it is insecure.

I know this makes it longer, but I think providing instructions on how to change the JSON is non-trivial and providing a secure example and an insecure example is better.

-d '{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"] }}'
----
+

> NOTE: If the resulting host registration command fails, you may require to add `--insecure` flag. It will then allow hosts to get registered with {Project} via {SmartProxy}.

Reviewer 2: ```suggestion
[NOTE]
====
If your hosts do not trust the SSL certificate of {ProjectServer}, add the `--insecure` flag to the registration command.
====
```

Author: Addressed :+1:

--user "_My_User_Name_" \
-H 'Content-Type: application/json'
----
+

> If the above host registration command fails, you may require to add `--insecure` flag.
> It will then allow hosts to get registered with {Project} via {SmartProxy}.

Reviewer: Can we use the same instruction as for the command line, with the exact instruction replaced?
```suggestion
If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `insecure` flag to the registration command.
```

Author: Added to CLI method as well :+1:

+
[options="nowrap" subs="+quotes,attributes"]
----
# curl -X POST https://{foreman-example-com}/api/registration_commands \
--user "_My_User_Name_" \

> -H 'Content-Type: application/json'

Reviewer 3: ```suggestion
-H 'Content-Type: application/json' \
```

> -d '{ "registration_command": { "insecure": true }}

Reviewer 3: ```suggestion
-d '{ "registration_command": { "insecure": true }}'
```

-d '{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"] }}'
----
+
If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `insecure` flag to the registration command.
+
[options="nowrap" subs="+quotes,attributes"]
----
# curl -X POST https://{foreman-example-com}/api/registration_commands \
--user "_My_User_Name_" \
-H 'Content-Type: application/json' \
-d '{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"], "insecure": true }}'
----
+
Use an activation key to simplify specifying the environments.
For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
endif::[]

> +
> ifeval::["{context}" == "load-balancing"]

Reviewer 2: ```suggestion
ifeval::["{context}" == "load-balancing"]
+
```

-d '{ "registration_command": { "activation_keys": ["_My_Activation_Key_1_, _My_Activation_Key_2_"] }}'
----
+

> If the above host registration command fails, you may require to add `--insecure` flag.
> It will then allow hosts to get registered with {Project} via {SmartProxy}.

Reviewer: And here
```suggestion
If your hosts do not trust the SSL certificate of {ProjectServer}, SSL validation can be disabled by adding the `insecure` flag to the registration command.
```

+
Include `{ "force": true }` to register a host that has been previously registered to a {SmartProxyServer}.
endif::[]

> 

Reviewer: I don't think this is correct since the diff of the rendered result now shows `+` before `Include` is rendered literally.

Author: Fixed. Aligned those paras to 1st point.

Keep in mind this can save the password in the shell history.
Alternatively, you can use a temporary personal access token instead of a password.
To generate a token in the {ProjectWebUI}, navigate to *My Account* > *Personal Access Tokens*.

> 

Reviewer 2: ```suggestion
```
